flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManager.java

```diff
@@ -499,6 +506,21 @@ public void execute(Runnable command) {
         }
     }
 
+    private class ResourceActionsImpl implements ResourceActions {
+        @Override
+        public void releaseResource(InstanceID instanceId, Exception cause) {
+            validateRunsInMainThread();
+
+            ActiveResourceManager.this.releaseResource(instanceId, cause);
+        }
+
+        @Override
+        public boolean allocateResource(WorkerResourceSpec workerResourceSpec) {
+            validateRunsInMainThread();
+            return startNewWorker(workerResourceSpec);
```

Comment:
`startNewWorker` is no longer needed for `ResourceManager` and `StandaloneResourceManager`.

Reply:
We probably have the same problem for `stopWorker`. There are some logics in `ResourceManager#releaseResource` that are not only for the active resource manager, however, the method `releaseResource` itself is never called except in the active resource manager.